src/plugin/memory-webview-main.ts

```diff
@@ -190,15 +189,18 @@ export class MemoryWebview implements vscode.CustomReadonlyEditorProvider {
             this.messenger.onRequest(logMessageType, message => outputChannelLogger.info('[webview]:', message), { sender: participant }),
             this.messenger.onRequest(readMemoryType, request => this.readMemory(request), { sender: participant }),
             this.messenger.onRequest(writeMemoryType, request => this.writeMemory(request), { sender: participant }),
-            this.messenger.onRequest(getVariables, request => this.getVariables(request), { sender: participant }),
+            this.messenger.onRequest(getVariablesType, request => this.getVariables(request), { sender: participant }),
             this.messenger.onNotification(resetMemoryViewSettingsType, () => this.setInitialSettings(participant, panel.title), { sender: participant }),
             this.messenger.onNotification(setTitleType, title => { panel.title = title; }, { sender: participant }),
+            this.messenger.onRequest(storeMemoryType, args => this.storeMemory(args), { sender: participant }),
+            this.messenger.onRequest(applyMemoryType, () => this.applyMemory(), { sender: participant }),
 
             this.memoryProvider.onDidStopDebug(() => {
                 if (this.refreshOnStop === RefreshEnum.on) {
                     this.refresh(participant);
                 }
             }),
+            this.memoryProvider.onDidWriteMemory(writtenMemory => this.messenger.sendNotification(memoryWrittenType, participant, writtenMemory))
```

Comment:
You are absolutely right! I was actually thinking about this as well but thought it might be too much for this PR. Currently we just notify all the memory inspectors and they'll reload their state if they match the changed range - even if they do not actually are about this session. Since this causes overhead due to more than necessary reads but not really any faults, thank you for creating a separate ticket for it!